hypershift-operator/controllers/hostedcluster/network_policies.go

```diff
@@ -698,3 +700,19 @@ func reconcileManagementKASNetworkPolicy(policy *networkingv1.NetworkPolicy, man
 
 	return nil
 }
+
+func detectSuboptimalMTU(hcluster *hyperv1.HostedCluster, managementClusterNetwork *configv1.Network) {
+	const recommendedMinMTU = 9000
+	if managementClusterNetwork.Status.ClusterNetworkMTU < recommendedMinMTU {
+		meta.SetStatusCondition(&hcluster.Status.Conditions, metav1.Condition{
```

Comment:
We need to ensure we're looking at the infra cluster.

expected MTU values when infra is 9000 MTU

9000 - hostnet infra
8900 - infra podnet
8900 - vm hostnet
8800 - vm podnet